File: sdk/chocobase-js/test/client.test.js
import test from 'node:test';
import assert from 'node:assert/strict';
import { createClient, ChocoClient } from '../dist/index.js';

test('createClient instantiates a valid ChocoClient', () => {
  const client = createClient('http://localhost:8080', 'test-anon-key');
  assert.ok(client instanceof ChocoClient);
  assert.ok(client.auth);
  assert.ok(client.storage);
  assert.ok(client.realtime);
  assert.ok(client.functions);
  assert.ok(client.graphql);
});

test('Query builder constructs correct REST URLs and headers', () => {
  const client = createClient('http://localhost:8080', 'test-anon-key');
  const query = client.from('users').select('id, name, email').eq('role', 'admin').limit(10);
  assert.ok(query);
});

test('Realtime client creates channels with event listeners', () => {
  const client = createClient('http://localhost:8080', 'test-anon-key');
  const channel = client.realtime.channel('chat_room');
  assert.ok(channel);
  let called = false;
  channel.on('message', (msg) => {
    called = true;
  });
  channel.unsubscribe();
});

File: sdk/chocobase-rs/tests/client_test.rs
use chocobase::create_client;

#[test]
fn test_rust_client_initialization() {
    let client = create_client("http://localhost:8080", "anon-key-123");
    assert_eq!(client.base_url, "http://localhost:8080");
    assert_eq!(client.api_key, "anon-key-123");

    let query = client.from("profiles").select("id, name").eq("role", "member").limit(10);
    assert_eq!(query.params.get("select").unwrap(), "id, name");
    assert_eq!(query.params.get("role").unwrap(), "eq.member");
    assert_eq!(query.params.get("limit").unwrap(), "10");

    let signed_url = client.storage.from("avatars").create_signed_url("user.png", 3600);
    assert!(signed_url.contains("/sign/user.png?expires_in=3600"));
}
